### PR TITLE
AT-4418 Make SDK tests blocking deploy

### PR DESCRIPTION
**Pull request by janchrizz**
We have modified the SDK tests [here](https://github.com/up42/sdk-tests/pull/28) to run the latest commit instead of latest release

This PR will make it blocking before deployment.

### CODE REVIEW

**Review by andher1802 — APPROVED** [review]
LGTM

**Review by seedlit — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -121,8 +122,6 @@ workflows:
           name: sdk-e2e-tests
           fingerprint: 'c5:5a:28:b3:5e:21:8d:9c:6c:b1:9b:a7:7b:f3:41:0c'
           run-command: poetry run pytest spec --junitxml=./report/junit.xml
-          requires:
-            - deploy
```
Comment (javidq) [inline .circleci/config.yml:125]: must depend on test

File: .circleci/config.yml [hunk 2]
```diff
@@ -121,8 +122,6 @@ workflows:
           name: sdk-e2e-tests
```
Comment (javidq) [inline .circleci/config.yml:122]: Let's move it up to be right after sonar

File: .circleci/config.yml [hunk 3]
```diff
@@ -112,6 +112,7 @@ workflows:
       - deploy:
           context: pypi-python-package-publishing
           requires:
+            - sdk-e2e-tests
```
Comment (javidq) [inline .circleci/config.yml:115]: Please move this to hold_before_deploy

**Review by javidq — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (janchrizz) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 2 above)
Comment (janchrizz) [reply .circleci/config.yml]: done

File: .circleci/config.yml [hunk 4]
```diff
@@ -102,10 +102,24 @@ workflows:
...
+          context:
+            - 'sdk-tests-staging-build'
+          post-steps:
+            - automated-test/notify-slack:
+                test-name: 'SDK E2E tests for up42-py failed!'
+                channel: *slack-channel-id-qa-alert
       - hold_before_deploy:
           type: approval
           requires:
             - test
```
Comment (javidq) [inline .circleci/config.yml:121]: Not needed anymore

**Review by javidq — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (janchrizz) [reply .circleci/config.yml]: done

**Review by javidq — APPROVED** [review]

File: .circleci/config.yml [hunk 5]
```diff
@@ -102,10 +102,23 @@ workflows:
           context: sonarcloud
           requires:
             - test
+      - automated-test/run-sdk-tests:
```
Comment (javidq) [inline .circleci/config.yml:105]: We need to restrict this to master branch only

**Review by javidq — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 5 above)
Comment (janchrizz) [reply .circleci/config.yml:105]: done as well

**Review by javidq — APPROVED** [review]

**Pull request merged**